### PR TITLE
Added Support for Java

### LINKED ISSUE #49: Add support for java
Need to add boilerplate generators

### PR DESCRIPTION
**Pull request by mohitejaikumar**
fix #49

### CODE REVIEW

**Comment by mohitejaikumar** [comment]
While execution function.java file locally change name of file to Main.java .

File: apps/problems/max-element/boilerplate-full/function.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+
+import java.io.*;
+import java.util.*;
+
+public class Main {
+    
+    ##USER_CODE_HERE##
+
+    public static void main(String[] args) {
```
Comment (SujithThirumalaisamy) [inline apps/problems/max-element/boilerplate-full/function.java:9]: The file path seems wrong. Can you make this like the other files do. It should be something like `/dev/problems`
That is the place where the problems are mounted

File: apps/boilerplate-generator/src/FullProblemDefinitionGenerator.ts [hunk 2]
```diff
@@ -87,6 +87,73 @@ int main() {
...
+    return `
+import java.io.*;
+import java.util.*;
+
+public class Main {
+    
+    ##USER_CODE_HERE##
+
+    public static void main(String[] args) {
+        String filePath = "/WebDevelopment/algorithmic-arena/apps/problems/${this.problemName.toLowerCase().replace(" ", "-")}/tests/inputs/##INPUT_FILE_INDEX##.txt";
```
Comment (SujithThirumalaisamy) [inline apps/boilerplate-generator/src/FullProblemDefinitionGenerator.ts:136]: Expected: `/dev/problems/${this.problemName.toLowerCase().replace(" ", "-")}/tests/inputs/##INPUT_FILE_INDEX##.txt`

Comment (mohitejaikumar) [reply apps/boilerplate-generator/src/FullProblemDefinitionGenerator.ts]: oh sorry , actually it was for local development ,
Changed it you can review

**Comment by mohitejaikumar** [comment]
Resolved some bug in generateCpp() of ProblemDefinationGenerator

File: apps/problems/classroom/boilerplate-full/function.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+
+import java.io.*;
+import java.util.*;
+
+public class Main {
+    
+    ##USER_CODE_HERE##
+
+    public static void main(String[] args) {
+        String filePath = "/WebDevelopment/algorithmic-arena/apps/problems/classroom/tests/inputs/##INPUT_FILE_INDEX##.txt";
```
Comment (siinghd) [inline apps/problems/classroom/boilerplate-full/function.java:10]: as suggest by @SujithThirumalaisamy this should be same as others

File: apps/problems/max-element/boilerplate-full/function.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        String filePath = "/WebDevelopment/algorithmic-arena/apps/problems/max-element/tests/inputs/##INPUT_FILE_INDEX##.txt";
```
Comment (siinghd) [inline apps/problems/max-element/boilerplate-full/function.java:10]: same here

File: apps/problems/two-sum/boilerplate-full/function.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
+
+import java.io.*;
+import java.util.*;
+
+public class Main {
+    
+    ##USER_CODE_HERE##
+
+    public static void main(String[] args) {
+        String filePath = "/WebDevelopment/algorithmic-arena/apps/problems/two-sum/tests/inputs/##INPUT_FILE_INDEX##.txt";
```
Comment (siinghd) [inline apps/problems/two-sum/boilerplate-full/function.java:10]: same here

**Review by siinghd — CHANGES_REQUESTED** [review]
Other than these small things looks fine

**Comment by mohitejaikumar** [comment]
Is there any bounty for this @hkirat

**Comment by mohitejaikumar** [comment]
is there any bounty for this @siinghd

**Comment by hkirat** [comment]
There would be one but we're giving them out together now at end of month after testing the issue
it'll prolly be ~$20

**Pull request merged**